### PR TITLE
Update options for `AnimationOptions`

### PR DESCRIPTION
**Pull request by cs09g**
- ~remove unsupported fields from easeTo's example~
- ~add optional types on flyTo's options type~
- add missing properties to the `AnimationOptions` docs

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [ ] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [ ] manually test the debug page
 - [ ] tagged `@mapbox/map-design-team` `@mapbox/static-apis` if this PR includes style spec API or visual changes
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [ ] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [ ] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog></changelog>`

### CODE REVIEW

**Comment by cs09g** [comment]
Unstable for camera test? It passes on local.

image

File: src/ui/camera.js [hunk 1]
```diff
@@ -1463,7 +1461,7 @@ class Camera extends Evented {
      * @see [Example: Slowly fly to a location](https://www.mapbox.com/mapbox-gl-js/example/flyto-options/)
      * @see [Example: Fly to a location based on scroll position](https://www.mapbox.com/mapbox-gl-js/example/scroll-fly-to/)
      */
-    flyTo(options: EasingOptions, eventData?: Object): this {
+    flyTo(options: EasingOptions & { curve?: number, minZoom?: number, speed?: number, screenSpeed?: number, maxDuration?: number }, eventData?: Object): this {
```
Comment (stepankuzmin) [inline src/ui/camera.js:1464]: Good catch, @cs09g!

These parameters should belong in the `AnimationOptions` type definition. Could you please move them into the `AnimationOptions` and revert the `speed` and `curve` deletion in the JSDoc example?

https://github.com/mapbox/mapbox-gl-js/blob/bf7f214a9bb48b55890313168a22ff8743cbdb51/src/ui/camera.js#L102-L127

Comment (cs09g) [reply src/ui/camera.js:1464]: @stepankuzmin One question.

These parameters are placed there as `@param` so the descriptions contain their default values now.
If these are moved to `AnimationOptions`, they would be placed as `@property` that doesn't contain its default value.

Is this okay to move these descriptions without modifying their contents?

Somewhat like as below:
```js
/**
 ...
 * @typedef {Object} AnimationOptions
 ...
 * @property {number} curve The zooming "curve" that will occur along the
 *     flight path. A high value maximizes zooming for an exaggerated animation, while a low
 *     value minimizes zooming for an effect closer to {@link Map#easeTo}. 1.42 is the average
 *     value selected by participants in the user study discussed in
 *     [van Wijk (2003)](https://www.win.tue.nl/~vanwijk/zoompan.pdf). A value of
 *     `Math.pow(6, 0.25)` would be equivalent to the root mean squared average velocity. A
 *     value of 1 would produce a circular motion. If `minZoom` is specified, this option will be ignored.
 ...
*/
```

Comment (stepankuzmin) [reply src/ui/camera.js:1464]: Good question! I think we can copy these descriptions without modifying their contents and still have the `@param` with the default value in the `flyTo` JSDoc.

Comment (cs09g) [reply src/ui/camera.js]: Thanks. I added properties to the `AnimationOptions` docs. Only removed `options.` on the descriptions.

**Review by stepankuzmin — APPROVED** [review]
Thank you, @cs09g. This looks good to me 👍

**Pull request merged**